package.json

```diff
@@ -39,18 +39,25 @@
     "grunt-eslint": "20.0.0",
     "grunt-git-authors": "3.2.0",
     "grunt-jsonlint": "1.1.0",
+    "grunt-karma": "2.0.0",
     "grunt-newer": "1.3.0",
     "grunt-npmcopy": "0.1.0",
     "gzip-js": "0.3.2",
     "husky": "0.14.3",
     "insight": "0.8.4",
     "jsdom": "5.6.1",
+    "karma": "1.7.0",
+    "karma-browserstack-launcher": "1.3.0",
+    "karma-chrome-launcher": "2.2.0",
+    "karma-firefox-launcher": "1.0.1",
+    "karma-qunit": "1.2.1",
     "load-grunt-tasks": "3.5.2",
     "native-promise-only": "0.8.1",
     "promises-aplus-tests": "2.1.2",
     "q": "1.5.0",
     "qunit-assert-step": "1.1.1",
-    "qunitjs": "2.4.0",
+    "qunitjs": "1.23.1",
```

Comment:
@mgol The previous upgrades were fake. We are still on QUnit 1.20, see `external` and note how the previous "updates" did not run npmcopy to update that file. Contrary to index.html, Karma does actually include qunitjs from node_modules directly and our tests don't actually support 2.x yet right now. This commit actually "upgrades" qunitjs from 1.20 to 1.23, although I haven't yet committed the updated external copy for the non-Karma test.

I'll do that in a separate commit before this one.